0.9.4/share/man/man1/ruby-install.1
.\" Generated by kramdown-man 0.1.8
.\" https://github.com/postmodern/kramdown-man#readme
.TH ruby-install 1 "Aug 2018" ruby-install "User Manuals"
.LP
.SH SYNOPSIS
.LP
.HP
\fBruby-install\fR \[lB]OPTIONS\[rB] \[lB]\[lB]RUBY \[or] VERSION \[or] RUBY\-VERSION\[rB] \[lB]\-\- CONFIGURE\[ru]OPTS\.\.\.\[rB]\[rB]
.LP
.SH DESCRIPTION
.LP
.PP
Installs Ruby, JRuby, TruffleRuby, or mruby\.
.LP
.PP
https:\[sl]\[sl]github\.com\[sl]postmodern\[sl]ruby\-install\[sh]readme
.LP
.SH ARGUMENTS
.LP
.TP
\fIRUBY\fP
Install Ruby by name\. Must be either \fBruby\fR, \fBjruby\fR, \fBtruffleruby\fR, or
\fBmruby\fR\.
.LP
.TP
\fIVERSION\fP
Optionally select the version of selected Ruby\.
.LP
.TP
\fICONFIGURE\[ru]OPTS\fP
Additional optional configure arguments\.
.LP
.SH OPTIONS
.LP
.TP
\fB-r\fR, \fB--rubies-dir\fR \fIDIR\fP
Specifies the alternate directory where other Ruby directories are
installed, such as \fI\[ti]\[sl]\.rvm\[sl]rubies\fP or \fI\[ti]\[sl]\.rbenv\[sl]versions\fP\.
Defaults to \fI\[sl]opt\[sl]rubies\fP for root and \fI\[ti]\[sl]\.rubies\fP for normal users\.
.LP
.TP
\fB-i\fR, \fB--install-dir\fR \fIDIR\fP
Specifies the directory where Ruby will be installed\.
Defaults to \fI\[sl]opt\[sl]rubies\[sl]\[Do]ruby\-\[Do]version\fP for root and
\fI\[ti]\[sl]\.rubies\[sl]\[Do]ruby\-\[Do]version\fP for normal users\.
.LP
.TP
\fB--prefix\fR \fIDIR\fP
Alias for \fB-i DIR\fR\.
.LP
.TP
\fB--system\fR
Alias for \fB-i $HOMEBREW_PREFIX\fR\.
.LP
.TP
\fB-s\fR, \fB--src-dir\fR \fIDIR\fP
Specifies the directory for downloading and unpacking Ruby source\.
.LP
.TP
\fB-c\fR, \fB--cleanup\fR
Remove the downloaded Ruby archive and unpacked source\-code after
installation\.
.LP
.TP
\fB-j\fR, \fB--jobs\fR \fINUM\fP, \fB-j\fR\fINUM\fP, \fB--jobs=\fR\fINUM\fP
Specifies the number of \fImake\fP jobs to run in parallel when compiling
Ruby\. If the \-j option is provided without an argument, \fImake\fP will
allow an unlimited number of simultaneous jobs\.
.LP
.TP
\fB-p\fR, \fB--patch\fR \fIFILE\fP
Specifies any additional patches to apply\.
.LP
.TP
\fB-M\fR, \fB--mirror\fR \fIURL\fP
Specifies an alternate mirror to download the Ruby archive from\.
.LP
.TP
\fB-u\fR, \fB--url\fR \fIURL\fP
Alternate URL to download the Ruby archive from\.
.LP
.TP
\fB-m\fR, \fB--md5\fR \fIMD5\fP
Specifies the MD5 checksum for the Ruby archive\.
.LP
.TP
\fB--sha1\fR \fISHA1\fP
Specifies the SHA1 checksum for the Ruby archive\.
.LP
.TP
\fB--sha256\fR \fISHA256\fP
Specifies the SHA256 checksum for the Ruby archive\.
.LP
.TP
\fB--sha512\fR \fISHA512\fP
Specifies the SHA512 checksum for the Ruby archive\.
.LP
.TP
\fB--package-manager [apt|dnf|yum|pacman|zypper|brew|pkg|port]\fR
Use an alternative package manager\.
.LP
.TP
\fB--no-download\fR
Use the previously downloaded Ruby archive\.
.LP
.TP
\fB--no-verify\fR
Do not verify the downloaded Ruby archive\.
.LP
.TP
\fB--no-extract\fR
Do not extract the downloaded Ruby archive\. Implies \fB--no-download\fR
and \fB--no-verify\fR\.
.LP
.TP
\fB--no-install-deps\fR
Do not install build dependencies before installing Ruby\.
.LP
.TP
\fB--no-reinstall\fR
Skip installation if another Ruby is detected in same location\.
.LP
.TP
\fB-U\fR, \fB--update\fR
Downloads the latest ruby versions and checksums from the ruby\-versions
repository (https:\[sl]\[sl]github\.com\[sl]postmodern\[sl]ruby\-versions\[sh]readme)\.
.LP
.TP
\fB-D\fR, \fB--debug\fR
Enable debugging messages\.
.LP
.TP
\fB-V\fR, \fB--version\fR
Prints the current ruby\-install version\.
.LP
.TP
\fB-h\fR, \fB--help\fR
Prints a synopsis of ruby\-install usage\.
.LP
.SH EXAMPLES
.LP
.PP
List supported Rubies and their major versions:
.LP
.nf
\[Do] ruby\-install
.fi
.LP
.PP
List the latest versions:
.LP
.nf
\[Do] ruby\-install \-\-latest
.fi
.LP
.PP
Install the current stable version of Ruby:
.LP
.nf
\[Do] ruby\-install ruby
.fi
.LP
.PP
Install the latest version of Ruby:
.LP
.nf
\[Do] ruby\-install \-\-latest ruby
.fi
.LP
.PP
Install a latest version of Ruby:
.LP
.nf
\[Do] ruby\-install ruby 3\.1
.fi
.LP
.PP
Install a specific version of Ruby:
.LP
.nf
\[Do] ruby\-install ruby 3\.1\.2
.fi
.LP
.PP
Install a Ruby into a specific directory:
.LP
.nf
\[Do] ruby\-install \-\-install\-dir \[sl]path\[sl]to\[sl]dir ruby
.fi
.LP
.PP
Install a Ruby into a specific \fBrubies\fR directory:
.LP
.nf
\[Do] ruby\-install \-\-rubies\-dir \[sl]path\[sl]to\[sl]rubies\[sl] ruby
.fi
.LP
.PP
Install a Ruby into \fB$HOMEBREW_PREFIX\fR:
.LP
.nf
\[Do] ruby\-install \-\-system ruby 3\.1\.2
.fi
.LP
.PP
Install a Ruby from an official site with directly download:
.LP
.nf
\[Do] ruby\-install \-M https:\[sl]\[sl]ftp\.ruby\-lang\.org\[sl]pub\[sl]ruby ruby 3\.1\.2
.fi
.LP
.PP
Install a Ruby from a mirror:
.LP
.nf
\[Do] ruby\-install \-M http:\[sl]\[sl]www\.mirrorservice\.org\[sl]sites\[sl]ftp\.ruby\-lang\.org\[sl]pub\[sl]ruby ruby 2\.4\.0
.fi
.LP
.PP
Install a Ruby with a specific patch:
.LP
.nf
\[Do] ruby\-install \-p https:\[sl]\[sl]raw\.github\.com\[sl]gist\[sl]4136373\[sl]falcon\-gc\.diff ruby 1\.9\.3\-p551
.fi
.LP
.PP
Install a Ruby with specific configuration:
.LP
.nf
\[Do] ruby\-install ruby 3\.1\.2 \-\- \-\-enable\-shared \-\-enable\-dtrace CFLAGS\[eq]\[dq]\-O3\[dq]
.fi
.LP
.PP
Using ruby\-install with \[lB]RVM\[rB]:
.LP
.nf
\[Do] ruby\-install \-\-rubies\-dir \[ti]\[sl]\.rvm\[sl]rubies ruby 3\.1\.2
.fi
.LP
.PP
Using ruby\-install with \[lB]rbenv\[rB]:
.LP
.nf
\[Do] ruby\-install \-i \[ti]\[sl]\.rbenv\[sl]versions\[sl]2\.4\.0 ruby 3\.1\.2
.fi
.LP
.PP
Uninstall a Ruby version:
.LP
.nf
\[Do] rm \-rf \[ti]\[sl]\.rubies\[sl]ruby\-3\.1\.2
.fi
.LP
.SH FILES
.LP
.TP
\fI\[sl]usr\[sl]local\[sl]src\fP
Default root user source directory\.
.LP
.TP
\fI\[ti]\[sl]src\fP
Default non\-root user source directory\.
.LP
.TP
\fI\[sl]opt\[sl]rubies\[sl]\[Do]ruby\-\[Do]version\fP
Default root user installation directory\.
.LP
.TP
\fI\[ti]\[sl]\.rubies\[sl]\[Do]ruby\-\[Do]version\fP
Default non\-root user installation directory\.
.LP
.SH ENVIRONMENT
.LP
.TP
\fIRUBY\[ru]INSTALL\[ru]SRC\[ru]DIR\fP
Overrides the default source directory\.
.LP
.TP
\fIRUBY\[ru]INSTALL\[ru]RUBIES\[ru]DIR\fP
Overrides the default installation directory\.
.LP
.TP
\fIRUBY\[ru]INSTALL\[ru]PKG\[ru]MANAGER\fP
Explicitly sets the package manager to use\.
.LP
.SH AUTHOR
.LP
.PP
Postmodern 
.MT postmodern\.mod3\[at]gmail\.com
.ME
.LP
.SH SEE ALSO
.LP
.PP
ruby(1), gem(1), chruby(1), chruby\-exec(1)


0.9.4/.brew/ruby-install.rb
class RubyInstall < Formula
  desc "Install Ruby, JRuby, Rubinius, TruffleRuby, or mruby"
  homepage "https://github.com/postmodern/ruby-install"
  url "https://github.com/postmodern/ruby-install/releases/download/v0.9.4/ruby-install-0.9.4.tar.gz"
  sha256 "c932e5a61e5cf11056ba6224fd75f75f2a71a991d4fc216d670ba96d7cc5c6a0"
  license "MIT"
  head "https://github.com/postmodern/ruby-install.git", branch: "master"

  depends_on "xz"

  def install
    system "make", "install", "PREFIX=#{prefix}"

    # Ensure uniform bottles across prefixes
    inreplace man1/"ruby-install.1", "/usr/local", "$HOMEBREW_PREFIX"
    inreplace [
      pkgshare/"ruby-install.sh",
      pkgshare/"truffleruby/functions.sh",
      pkgshare/"truffleruby-graalvm/functions.sh",
    ], "/usr/local", HOMEBREW_PREFIX
  end

  test do
    system bin/"ruby-install"
  end
end
